In [ ]:
import numpy as np
import pandas as pd

In [ ]:
data=[
      [50,70],
      [80,100],
      [60,90],
      [30,50],
      [20,50]
]

'''
Satisfaction	Smart Thermostat	Smart Light	Total
Very Satisfied	50	70	120
Satisfied	80	100	180
Neutral	60	90	150
Unsatisfied	30	50	80
Very Unsatisfied	20	50	70
Total	240	360	600
'''

'\nSatisfaction\tSmart Thermostat\tSmart Light\tTotal\nVery Satisfied\t50\t70\t120\nSatisfied\t80\t100\t180\nNeutral\t60\t90\t150\nUnsatisfied\t30\t50\t80\nVery Unsatisfied\t20\t50\t70\nTotal\t240\t360\t600\n'

In [ ]:
data

[[50, 70], [80, 100], [60, 90], [30, 50], [20, 50]]

In [ ]:
# df=pd.DataFrame(data)
# df

In [ ]:
from scipy.stats import chi2_contingency

In [ ]:
chi2_contingency(data)

Chi2ContingencyResult(statistic=5.638227513227513, pvalue=0.22784371130697179, dof=4, expected_freq=array([[ 48.,  72.],
       [ 72., 108.],
       [ 60.,  90.],
       [ 32.,  48.],
       [ 28.,  42.]]))

In [ ]:
calc_stat,p_value,dof,expected=chi2_contingency(data)

In [ ]:
alpha=0.05
if p_value<=alpha:
  print('Reject Null')
else:
  print('Accept Null')

Accept Null


# There's a no significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.

In [ ]:
# at dof=4, alpha=0.05
#from chi square distribution table
Critical_chi_stat=9.488

As critical value (9.488) > calculated value(5.638)
# Accept null hypothesis
There's a no significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.